File: data structures and algorithms/java/05_052.java
import java.util.*;
import java.util.stream.*;
import java.io.*;
import java.math.*;

public class Main {
    public static void main(String[] args) {
        // Main method code goes here
    }
    
    public class Node {
        public int key;
        public Node left;
        public Node right;
        
        public Node(int key) {
            this.key = key;
        }
    }
    
    public int kthSmallest(Node root, int k) {
        // Perform inorder traversal and return the kth element
        List<Integer> result = new ArrayList<>();
        
        inorder(root, result, k);
        
        return k <= result.size() ? result.get(k-1) : -1;
    }
    
    private void inorder(Node node, List<Integer> result, int k) {
        if (node == null || result.size() >= k) {
            return;
        }
        inorder(node.left, result, k);
        result.add(node.key);
        inorder(node.right, result, k);
    }
}

File: data structures and algorithms/cs/05_052.cs
using System;
using System.Collections.Generic;

public class MainClass
{
   public static void Main(string[] args)
   {
      // Main method code goes here
   }

   public class Node
   {
      public int Key;
      public Node Left;
      public Node Right;

      public Node(int key)
      {
         this.Key = key;
      }
   }

   public int KthSmallest(Node root, int k)
   {
      // Perform inorder traversal and return the kth element
      List<int> result = new List<int>();

      Inorder(root, result, k);

      return k <= result.Count ? result[k - 1] : -1;
   }

   private void Inorder(Node node, List<int> result, int k)
   {
      if (node == null || result.Count >= k)
      {
         return;
      }
      Inorder(node.Left, result, k);
      result.Add(node.Key);
      Inorder(node.Right, result, k);
   }
}
